### PR TITLE
Fireblocks Integration: V3

### PR DESCRIPTION
**Pull request by rkharevych**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

SDFSOF-22: Process inbound payment (Withdrawal) via FB
SDFSOF-23: Process outbound payment (Deposit) via FB
SDFSOF-31: Integration, end to end tests
SDFSOF-25: Add reconciliation logic

### Why

Contains changes for Fireblocks webhooks handling, reconcilation job logic and integration tests

### CODE REVIEW

**Comment by Ifropc** [comment]
Could you please create another PR that merges develop -> softserve_develop?
Then, merge softserve_develop -> here.
It looks like this branch is quite out of date with develop.

**Comment by rkharevych** [comment]
> Could you please create another PR that merges develop -> softserve_develop? Then, merge softserve_develop -> here. It looks like this branch is quite out of date with develop.

Hi, Gleb.
We pulled changes from `softserve_develop` and added End2End tests with enabled custody integration. It can be easily configured using test profile. Also added Fireblocks "no fullstack" and "with fullstack" run configurations. To run Fireblocks End2End tests you need to provide the following environment variables:
- CUSTODY_TYPE=fireblocks
- CUSTODY_FIREBLOCKS_VAULT_ACCOUNT_ID=<your_value>
- SECRET_CUSTODY_FIREBLOCKS_API_KEY=<your_value>
- SECRET_CUSTODY_FIREBLOCKS_SECRET_KEY=<your_value>

Fireblocks End2End tests use a reconciliation job to update payment status and info instead of webhooks. It is quite difficult to configure webhooks because the webhook endpoint should be exposed to the Internet so that Fireblocks can send events to it. It can be done using something like `ngrok`, but I think, that integration tests cover webhooks functionality completely

File: platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPayment.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    String transactionEnvelope = null;
+
+    if (paymentOperation.isPresent()) {
+      id = paymentOperation.get().getId().toString();
+      to = paymentOperation.get().getTo();
+      amount = paymentOperation.get().getAmount();
+      assetType = paymentOperation.get().getAsset().getType();
+      assetName = paymentOperation.get().getAsset().toString();
+
+      if (paymentOperation.get().getAsset() instanceof AssetTypeCreditAlphaNum) {
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPayment.java:72]: Can you add support for native (xlm) type?

**Comment by rkharevych** [comment]
> > Could you please create another PR that merges develop -> softserve_develop? Then, merge softserve_develop -> here. It looks like this branch is quite out of date with develop.
> 
> Hi, Gleb. We pulled changes from `softserve_develop` and added End2End tests with enabled custody integration. It can be easily configured using test profile. Also added Fireblocks "no fullstack" and "with fullstack" run configurations. To run Fireblocks End2End tests you need to provide the following environment variables:
> 
> * CUSTODY_TYPE=fireblocks
> * CUSTODY_FIREBLOCKS_VAULT_ACCOUNT_ID=<your_value>
> * SECRET_CUSTODY_FIREBLOCKS_API_KEY=<your_value>
> * SECRET_CUSTODY_FIREBLOCKS_SECRET_KEY=<your_value>
> 
> Fireblocks End2End tests use a reconciliation job to update payment status and info instead of webhooks. It is quite difficult to configure webhooks because the webhook endpoint should be exposed to the Internet so that Fireblocks can send events to it. It can be done using something like `ngrok`, but I think, that integration tests cover webhooks functionality completely.
> 
> Tomorrow we will add one more thing - possibility to run Custody End2End tests with `./gradlew testFireblocksE2E`

Command to run Fireblocks End2End tests: 
`CUSTODY_TYPE=fireblocks CUSTODY_FIREBLOCKS_VAULT_ACCOUNT_ID=<your_value> SECRET_CUSTODY_FIREBLOCKS_API_KEY=<your_value> SECRET_CUSTODY_FIREBLOCKS_SECRET_KEY=<your_value> ./gradlew clean testFireblocksE2E`

**Pull request merged**